File: app/src/main/java/com/otoniel/alura/model/Student.java
package com.otoniel.alura.model;

import androidx.annotation.NonNull;

import java.io.Serializable;

public class Student implements Serializable {
    final private int id;
    private String nome;
    private String phone;
    private String email;

    public Student(int id, String nome, String phone, String email) {
        this.id = id;
        this.email = email;
        this.nome = nome;
        this.phone = phone;
    }

    public Student() {
        this.id = 0;
    }

    public int getId() {
        return id;
    }

    public String getNome() {
        return nome;
    }

    public String getPhone() {
        return phone;
    }

    public String getEmail() {
        return email;
    }

    public void setNome(String nome) {
        this.nome = nome;
    }

    public void setPhone(String phone) {
        this.phone = phone;
    }

    public void setEmail(String email) {
        this.email = email;
    }

    @Override
    @NonNull
    public String toString() {
        return "Student{" +
                "id='" + id + '\'' +
                "nome='" + nome + '\'' +
                ", phone='" + phone + '\'' +
                ", email='" + email + '\'' +
                '}';
    }
}

File: app/src/main/java/com/otoniel/alura/model/Student.kt
package com.otoniel.alura.model

import androidx.room.Entity
import androidx.room.Ignore
import androidx.room.PrimaryKey
import java.io.Serializable
import java.util.*

@Entity
class Student : Serializable {
    @Ignore
    constructor(id: Int, nome: String, phone: String, email: String, lastName: String, createdAt: Date?) {
        this.id = id
        this.email = email
        this.name = nome
        this.phone = phone
        this.lastName = lastName
        this.createdAt = createdAt
    }

    constructor()

    @PrimaryKey(autoGenerate = true)
    var id: Int = 0
    var name: String = ""
    var phone: String = ""
    var email: String = ""
    var lastName: String = ""
    var createdAt: Date? = Calendar.getInstance().time

    val fullName: String
        get() {
            return name + lastName
        }

    override fun toString(): String {
        return "Student{" +
                "id='" + id + '\'' +
                "name='" + name + '\'' +
                ", phone='" + phone + '\'' +
                ", email='" + email + '\'' +
                ", lastName='" + lastName + '\'' +
                '}'
    }
}